File: diss/code_snippets/coin.ml
let coin_mean inference_algorithm =
  let coin heads =
    let* theta = continuous_uniform 0. 1. in
    observe heads (binomial 10 theta) (return theta) in
  let posterior_single_coin = infer (coin 9) inference_algorithm in
  sample_mean ~n:10000 posterior_single_coin

(* 0.833 *)

File: code_snippets/webppl/coin.js
var coin_model = function (method) {
  var coin = function () {
    var theta = uniform(0.0, 1.0);
    observe(
      Binomial({ n: 10, p: theta }),9
    );
    return theta;
  };
  return Infer(method, coin);
};
